src/software/ai/passing/pass_generator.cpp

```diff
@@ -106,24 +104,16 @@ PassGenerator::~PassGenerator()
 
 void PassGenerator::continuouslyGeneratePasses()
 {
-    // Take ownership of the in_destructor flag so we can use it for the conditional
-    // check
-    in_destructor_mutex.lock();
-    while (!in_destructor)
+    while (!in_destructor.load())
     {
-        // Give up ownership of the in_destructor flag now that we're done the
-        // conditional check
-        in_destructor_mutex.unlock();
-
-        updateAndOptimizeAndPrunePasses();
+        if (pass_generator_running.load())
```

Comment:
If the pass generator is stopped, any idea what the CPU load is for an empty loop with a `yield`? Just seems kinda odd although I'm not sure how much CPU it will take up. Perhaps `start` and `stop` should start or destroy the separate thread? Or would that not work with the "shared" interface between deterministic and non-deterministic mode?

Reply:
Mostly bringing this up because IIRC the main motivation for adding `start/stop` was to reduce the load on the CPU when we didn't need the pass generator to be running, so it woulnd't run for the entire duration of a play.

Reply:
What if I can achieve the same CPU performance improvements just by making the pass generator a local variable that gets deleted as soon as it's not needed?

Reply:
That's an interesting idea, although would require one more level of nesting in plays (to enclose in `{}`). Start/stop may be more intuitive to use.

@garethellis0 any other pros/cons for scoped desctruction vs start/stop?

Reply:
plays are already unwieldy so they would benefit from the refactoring